open_schools_platform/organization_management/teachers/selectors.py

```diff
@@ -24,7 +38,11 @@ def get_teacher_profile(*, filters=None, user: User = None) -> QuerySet:
     qs = TeacherProfile.objects.all()
     teacher_profile = TeacherProfileFilter(filters, qs).qs.first()
 
-    if user and teacher_profile and not user.has_perm('students.teacher_profile_access', teacher_profile):
+    if user and teacher_profile and not user.has_perm('teachers.teacher_profile_access', teacher_profile):
         raise PermissionDenied
 
     return teacher_profile
+
+
+def get_teachers_by_circles(circles: QuerySet) -> QuerySet:
+    return get_teachers(filters={"circle_ids": form_ids_string_from_queryset(circles)})
```

Comment:
why we need to use selector here
circle has one teacher, can we use attribute of circle there?

Reply:
Yes, I think u are right, we can use circle.teachers attr here.

But still, does circle has only 1 teacher? Because our architecture allows many teachers for 1 circle. For example, we have 3 Teacher's and they all are connected to 1 circle. And if we will call circle.teachers for this object - we will get 3 Teacher's.

Reply:
It is a product question. Okay let's leave this question for further time